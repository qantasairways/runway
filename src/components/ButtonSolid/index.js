import React from 'react';
import ButtonBase, {
  buttonPropTypes,
  buttonDefaultProps
} from '../../baseComponents/Button';
import { CSS_SELECTOR_ACTIVE, CSS_SELECTOR_HOVER } from '../../constants/css';

const themeStyles = {
  button: theme => ({
    backgroundColor: theme.colours.primary,
    color: theme.colours.white,
    [CSS_SELECTOR_HOVER]: {
      backgroundColor: theme.colours.primaryDark
    },
    [CSS_SELECTOR_ACTIVE]: {
      backgroundColor: theme.colours.primaryDark
    }
  })
};

const ButtonSolid = props => (
  <ButtonBase {...props} themeStyles={themeStyles} />
);

ButtonSolid.propTypes = {
  ...buttonPropTypes
};

ButtonSolid.defaultProps = {
  ...buttonDefaultProps
};

export default ButtonSolid;
