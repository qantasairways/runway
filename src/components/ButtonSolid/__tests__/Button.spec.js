import React from 'react';
import ButtonSolid from '..';
import { shallow } from 'enzyme';

describe('ButtonSolid', () => {
  it('should render correctly with defaults', () => {
    const component = shallow(<ButtonSolid />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const component = shallow(
      <ButtonSolid type="submit" onClick={() => 1}>
        button text
      </ButtonSolid>
    );

    expect(component).toMatchSnapshot();
  });
});
