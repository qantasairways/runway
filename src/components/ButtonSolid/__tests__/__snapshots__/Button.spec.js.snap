// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSolid should render correctly with defaults 1`] = `
<WithTheme(ThemedButton)
  disabled={false}
  inline={false}
  onClick={[Function]}
  themeStyles={
    Object {
      "button": [Function],
    }
  }
  type="button"
>
  button
</WithTheme(ThemedButton)>
`;

exports[`ButtonSolid should render correctly with props 1`] = `
<WithTheme(ThemedButton)
  disabled={false}
  inline={false}
  onClick={[Function]}
  themeStyles={
    Object {
      "button": [Function],
    }
  }
  type="submit"
>
  button text
</WithTheme(ThemedButton)>
`;
